workflow/executor/docker/docker.go

```diff
@@ -185,6 +185,8 @@ func (d *DockerExecutor) GetExitCode(ctx context.Context, containerName string)
 }
 
 func (d *DockerExecutor) Wait(ctx context.Context, containerNames []string) error {
+	ctx, cancel := context.WithCancel(ctx) // stop the polling when we are no longer waiting
+	defer cancel()
 	go func() {
 		err := d.pollContainerIDs(ctx, containerNames)
```

Comment:
If `pollContainerIDs` returns with a timeout error, it could stay in `for !d.haveContainers(containerNames)` forever.

Reply:
poll now runs for ever